cuqi/implicitprior/_regularizedGaussian.py

```diff
@@ -206,16 +256,18 @@ def sqrtcov(self):
     def sqrtcov(self, value):
         self.gaussian.sqrtcov = value     
     
-    def get_conditioning_variables(self):
-        return self.gaussian.get_conditioning_variables()
-    
     def get_mutable_variables(self):
-        return self.gaussian.get_mutable_variables()
+        add = []
+        if self.preset in self.regularization_options():
+            add = ["strength"]
+        return self.gaussian.get_mutable_variables() + add
     
     # Overwrite the condition method such that the underlying Gaussian is conditioned in general, except when conditioning on self.name
     # which means we convert Distribution to Likelihood or EvaluatedDensity.
     def _condition(self, *args, **kwargs):
-
+        if self.preset in self.regularization_options():
+            return super()._condition(*args, **kwargs)
```

Comment:
There seem to be some potential issues here. Namely, what if both the mean and strength were conditioning variables? Then we would only condition on the strength now. E.g.
```python
x = RegularizedGaussian(lambda x: x, 1, strength=lambda s: s, ...) 
x(strength=1, mean=2)
```
I suggest moving this check after creating "new_density" at the end, then doing the super()._condition(*args, **kwargs) on new_density.

It could also be worthwhile to add a test on this new conditioning feature.

Reply:
The precise workings of the conditioning is still a little bit of a mystery to me, but the issue you suggest, where there are anonymous functions for both Gaussian parameters and regularization parameters, seems to be working just fine. Also, in the full version of the code that I have, including conjugacy, this seems to work as well. 

![image](https://github.com/user-attachments/assets/12abc9b8-133e-47d3-aea1-0247f7459ba0)



I do get that this is not the best solution, however your suggestion,
if I interpret it as removing the initial check and replacing the final line with
`return new_density._condition(**kwargs)` crashes the kernel.

I have now added two tests, one with constraints and one with tv, to test the mutable and conditioning variables and verify that the conditioning does work as expected.

Reply:
Thank you. I look into it a bit more. It seems to be working as expected though